techmatters/terraso-mobile-client
# fix: info overlay sheet crashes

## Description

Fixes crash issues with info overlay sheets opening on certain screens.

This was inadvertently introduced by #1397, which enabled the Gorhom Bottom Sheet Modal's nested scrolling; however, this also introduced a dependency on having a navigation context available. Because our bottom sheet providers were not consistently placed in the application hierarchy, certain screens did not have access to the context, which caused the crash. This fix standardizes their provisioning to occur within the `ScreenScaffold` component and ensure that all bottom sheets exist within it. Incidentally, this also fixes issues with sheets opening to unexpected heights, which was another side-effect of the inconsistent placement of the `Provider` component.

### Checklist

### Related Issues
Fixes #1533

### Verification steps

Visit all screens featuring info overlay sheets. Open them and verify that the application does not crash.
